hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerProtocolServerSideTranslatorPB.java

```diff
@@ -115,7 +119,36 @@ public OMResponse submitRequest(RpcController controller,
           }
         }
       } else {
-        return submitRequestDirectlyToOM(request);
+        OMClientResponse omClientResponse = null;
+        long index = 0L;
+        try {
+          OMClientRequest omClientRequest =
+              OzoneManagerRatisUtils.createClientRequest(request);
+          if (omClientRequest != null) {
+            request = omClientRequest.preExecute(ozoneManager);
+            index = transactionIndex.incrementAndGet();
+            omClientRequest =
+                OzoneManagerRatisUtils.createClientRequest(request);
+            omClientResponse =
+                omClientRequest.validateAndUpdateCache(ozoneManager, index,
+                    ozoneManagerDoubleBuffer::add);
+          } else {
+            return submitRequestDirectlyToOM(request);
+          }
+        } catch(IOException ex) {
+          // As some of the preExecute returns error. So handle here.
+          return createErrorResponse(request, ex);
+        }
+
+        try {
+          omClientResponse.getFlushFuture().get();
+          LOG.trace("Future for {} is completed", request);
+        } catch (ExecutionException | InterruptedException ex) {
+          // Do we need to terminate OM here?
```

Comment:
The only potential issue is if we update the cache but fail to update RocksDB. In that case self-terminate is likely the best option. Since Ratis is not enabled the request is not persistent.

Reply:
So, terminate OM here?

Reply:
Yes I would err on the side of safety and self-terminate the OM here.